### DISCUSSION TITLE
Dynamic text content in theme.rasi

### DISCUSSION THREAD

**Discussion by Passeriform** [davatorium/rofi#1637] [Ideas/Feature Requests]
It'd be nice to have dynamic content string for textbox/buttons fetched from a script. It could be as basic as re-echo-ing whatever the script echoes.

Usage could look something like:

```
#textbox-running-a-script {
    content: "./script.sh";
}
```

script.sh
```sh
#!/bin/env bash
SOME_CONDITION=1

if $SOME_CONDITION; then
    echo "Positive"
else
    echo "Negative"
fi
```

Or if we don't want to disturb existing `content` property, something like this could work:

```
#textbox-running-a-script {
    content-from: "./script.sh";
}
```

**Comment by DaveDavenport** [comment]
I don't think this is a feature I want to see added to rofi. 

What are you trying to achieve high-level that requires this feature?

**Reply by Passeriform** [reply]
I'm currently trying to make an applet for wifi, bluetooth, etc, which I'm planning to have a __list of options__ (currently using rofi's -dmenu option) and a __bunch of buttons__ (mainly to provide toggles).

I was able to setup the buttons as a custom container widget with button widgets in the mainbox children packing. But the issue now remains that I want to show content of those buttons based on the status coming from a script.

Something like if bluetooth is on, the bluetooth toggle should say `Turn off` else it should be `Turn on`.

I am currently using a simple workaround by calling the button as `Toggle bluetooth` and showing status in `message` (rofi's -mesg option), which works but honestly is a little unintuitive.

PS: I'm coming from polybar so please forgive me if my mindset doesn't align well with Rofi ideology

**Reply by DaveDavenport** [reply]
You can achieve what you want using environment variables. 
For strings your environment variable value needs to have opening and closing ".

**Reply by Passeriform** [reply]
Ah, of course. That worked. Just needed to add extra quotes around my string.

Thanks @DaveDavenport
